bentoml/_internal/frameworks/catboost.py

```diff
@@ -0,0 +1,276 @@
+            Either the tag of the model to get from the store, or a BentoML `~bentoml.Model`
+            instance to load the model from.
+    Returns:
+        :obj:`~catboost.CatBoost`: The CatBoost model loaded from the model store or BentoML :obj:`~bentoml.Model`.
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+        # target model must be from the BentoML model store
+        booster = bentoml.catboost.load_model("my_catboost_model")
+    """  # noqa: LN001
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = get(bento_model)
+
+    if bento_model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, not loading with {MODULE_NAME}."
+        )
+
+    model_file = bento_model.path_of(MODEL_FILENAME)
+    cb_class_name: str = bento_model.info.options.training_class_name or DEFAULT_MODEL_TRAINING_CLASS_NAME  # type: ignore
+    cb_class: t.Type[cb.CatBoost] = getattr(cb, cb_class_name, cb.CatBoost)
+    if not issubclass(cb_class, cb.CatBoost) and cb_class is not cb.CatBoost:
+        raise BentoMLException(f"{cb_class_name} is not a valid CatBoost class.")
+    cb_instance = cb_class()
+    booster: cb.CatBoost = cb_instance.load_model(fname=model_file)
+    return booster
+
+
+@attr.define
+class CatBoostOptions(ModelOptions):
+    training_class_name: str = attr.field(factory=str)
+
+
+def save_model(
+    name: str,
+    model: cb.CatBoost,
+    *,
+    signatures: dict[str, ModelSignatureDict] | None = None,
+    labels: dict[str, str] | None = None,
+    custom_objects: dict[str, t.Any] | None = None,
+    metadata: dict[str, t.Any] | None = None,
+) -> bentoml.Model:
+    """
+    Save an CatBoost model instance to the BentoML model store.
+
+    Args:
+        name:
+            The name to give to the model in the BentoML store. This must be a valid
+            :obj:`~bentoml.Tag` name.
+        model:
+            The CatBoost model to be saved.
+        signatures:
+            Signatures of predict methods to be used. If not provided, the signatures default to
+            ``{"predict": {"batchable": False}}``. See :obj:`~bentoml.types.ModelSignature` for more
+            details.
+        labels:
+            A default set of management labels to be associated with the model. An example is
+            ``{"training-set": "data-1"}``.
+        custom_objects:
+            Custom objects to be saved with the model. An example is
+            ``{"my-normalizer": normalizer}``.
+
+            Custom objects are currently serialized with cloudpickle, but this implementation is
+            subject to change.
+        metadata:
+            Metadata to be associated with the model. An example is ``{"max_depth": 2}``.
+
+            Metadata is intended for display in model management UI and therefore must be a default
+            Python type, such as ``str`` or ``int``.
+    Returns:
```

Comment:
@aarnphm Still the same question, do we need to modify all the framework documentation? Because almost all the documentation has the problems you pointed out

Reply:
we can have a PR to address this all at once. This is due to our lack of doctest. Will propose a way to do this after 1.0

Reply:
I will resolve this for now. cc @aarnphm